File: ContactList/app/src/main/java/com/bergamin/contactlist/adapter/ContactsAdapter.java
package com.bergamin.contactlist.adapter;

import android.content.Context;
import android.graphics.Bitmap;
import android.graphics.BitmapFactory;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.ImageView;
import android.widget.TextView;

import com.bergamin.contactlist.ContactListActivity;
import com.bergamin.contactlist.R;
import com.bergamin.contactlist.model.Contact;

import java.util.List;

/**
 * Created by Guilherme on 20/08/2016.
 */
public class ContactsAdapter extends BaseAdapter {
    private final List<Contact> contacts;
    private final Context context;

    public ContactsAdapter(Context context, List<Contact> contacts) {
        this.context = context;
        this.contacts = contacts;
    }

    @Override
    public int getCount() {
        return contacts.size();
    }

    @Override
    public Object getItem(int position) {
        return contacts.get(position);
    }

    @Override
    public long getItemId(int position) {
        return contacts.get(position).getId();
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {

        Contact contact = contacts.get(position);
        LayoutInflater inflater = LayoutInflater.from(context);

        if(convertView == null) {
            convertView = inflater.inflate(R.layout.list_item, null);
        }

        TextView nameTxt = (TextView) convertView.findViewById(R.id.item_name);
        TextView phoneTxt = (TextView) convertView.findViewById(R.id.item_phone);
        ImageView photoImg = (ImageView) convertView.findViewById(R.id.item_photo);

        nameTxt.setText(contact.getName());
        phoneTxt.setText(contact.getPhone());

        String filePath = contact.getPhotoPath();
        if(filePath != null) {
            Bitmap bitmap = BitmapFactory.decodeFile(filePath);
            bitmap = Bitmap.createScaledBitmap(bitmap, 100, 100, true);
            photoImg.setImageBitmap(bitmap);
            photoImg.setScaleType(ImageView.ScaleType.FIT_XY);
        }

        return convertView;
    }
}

File: Kotlin/ContactList/app/src/main/java/com/bergamin/contactlist/adapter/ContactsAdapter.kt
package com.bergamin.contactlist.adapter

import android.content.Context
import android.graphics.Bitmap
import android.graphics.BitmapFactory
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import android.widget.ImageView
import android.widget.TextView
import com.bergamin.contactlist.R
import com.bergamin.contactlist.model.Contact

/**
 * Created by Guilherme Taffarel Bergamin on 28/11/2017.
 */
class ContactsAdapter(
        private val context: Context,
        private val contacts: List<Contact>?): BaseAdapter() {

    override fun getView(position: Int, convertView: View?, parent: ViewGroup?): View {
        var contact = contacts!![position]
        var inflater = LayoutInflater.from(context)
        var view = convertView

        if(convertView == null){
            view = inflater.inflate(R.layout.list_item,null)
        }

        var itemName: TextView? = view?.findViewById(R.id.itemName)
        var itemPhone: TextView? = view?.findViewById(R.id.itemPhone)
        var itemPhoto: ImageView? = view?.findViewById(R.id.itemPhoto)
        var itemAddress: TextView? = view?.findViewById(R.id.itemAddress)
        var itemWebSite: TextView? = view?.findViewById(R.id.itemWebSite)

        itemName?.text = contact.name
        itemPhone?.text = contact.phone
        if(itemAddress != null) itemAddress.text = contact.address
        if(itemWebSite != null) itemWebSite.text = contact.webSite

        if(contact.photoPath != "") {
            var bitmap = BitmapFactory.decodeFile(contact.photoPath)
            bitmap = Bitmap.createScaledBitmap(bitmap, 100, 100, true)
            itemPhoto?.setImageBitmap(bitmap)
            itemPhoto?.scaleType = ImageView.ScaleType.FIT_XY
        }
        return view!!
    }

    override fun getItem(position: Int): Any {
        return contacts!![position]
    }

    override fun getItemId(position: Int): Long {
        return contacts!![position].id
    }

    override fun getCount(): Int {
        return contacts?.size ?: 0
    }
}
